python-lib/results/ab_statistics.py

```diff
@@ -0,0 +1,52 @@
+import pandas as pd
+
+from constants import Group, Columns
+
+
+class AbStatistics(object):
+    def __init__(self, user_reference_column: str, group_column: str, conversion_column: str):
+        self.user_reference_column = user_reference_column
+        self.group_column = group_column
+        self.conversion_column = conversion_column
+
+    def compute(self, results_df):
+        results_df = self.check_results_df(results_df)
+        aggregation = self.aggregate_by_group(results_df)
+        statistics_df = self.format_statistics(aggregation)
+        return statistics_df
+
+    def check_results_df(self, result_df):
+        valid_rows = result_df[[self.user_reference_column, self.group_column, self.conversion_column]].dropna()
+        not_empty_rows_nb = valid_rows.shape[0]
+        if not result_df[self.user_reference_column].is_unique:
+            raise ValueError("There should be only one row per user in the input dataset")
+        if not_empty_rows_nb < 2:
+            raise ValueError("The input dataset should contain at least 2 users with conversion and group references")
+        invalid_groups = valid_rows[(valid_rows[self.group_column] != Group.A.value) & (valid_rows[self.group_column] != Group.B.value)]
+        if not invalid_groups.empty:
+            raise ValueError("The group indicator should be either 'A' or 'B'")
+        invalid_conversion = valid_rows[(valid_rows[self.conversion_column] != 0) & (valid_rows[self.conversion_column] != 1)]
+        if not invalid_conversion.empty:
+            raise ValueError("The group indicator should be either 0 or 1")
+        return valid_rows
+
+    def aggregate_by_group(self, result_df):
+        aggregation = result_df.groupby([self.group_column]).agg({self.user_reference_column: "nunique", self.conversion_column: "sum"})
+        return aggregation
+
+    def format_statistics(self, aggregation):
+        groups = [Group.A.value, Group.B.value]
+        rows = []
+        for group in groups:
+            rows.append(self.statistic_per_group(group, aggregation))
+        statistics_df = pd.DataFrame(rows)
+        return statistics_df
+
+    def statistic_per_group(self, group, aggregation):
+        group_aggregation = aggregation.loc[group]
+        size = group_aggregation[self.user_reference_column]
+        row = {Columns.AB_GROUP: group, Columns.SUCCESS_RATE: self.conversion_rate(group_aggregation, size), Columns.SIZE: size}
```

Comment:
it does not make sense for the user that the AB reference column is always `dku_ab_group`, it should be whatever column name the users choses in the recipe param

Reply:
I think I chose a standard name because I wanted to make it easier for the web app which takes this dataset as input. From this dataframe, the web app retrieves the sample sizes and the success rates for each group. 
So if I let the user decides on the column names, it adds another parameter to the web app. I guess that's fine.

Actually, what would be perfect for the result web app would be to have the same kind of UI as in the visual statistic tool : 
1) pick the group column
2) pick the groups that you want to compare (A and B or A_group and B_group)

I guess that's feasible with 2 custom selects that retrieves the group values

Reply:
done in https://github.com/dataiku/dss-plugin-ab-testing/pull/2/commits/d3b6ebc0487f2213eef0edcfd63fd6bba5e990b4